### DISCUSSION TITLE
建议：全新的世界，考虑使用全新的文件格式 .xmd

### DISCUSSION THREAD

**Discussion by chuxinyuan** [yihui/litedown#43] [Ideas]
1. yaml 里的关键词后面的冒号输入完毕后要是有一个下拉框菜单提示就好了，但是涉及的 .Rmd 文档，情况可能复杂点，如果是有一个全新的文档格式，没有了历史包袱，定义域会小很多，实现起来是不是就容易多了。
2. 包名字改为 **xdown**.
3. RStudio 里新建文档那里，From Template 里建议安排上 **xdown** 的席位。
4. 帮助文档小品文里是否可以提供类似 [quarto reference](https://quarto.org/docs/reference/formats/html.html) 这样的文档，自定义起来会轻松很多。
5. 作为传递信息的介质，需且仅需一个。输出格式里建议有且仅有 html 即可，没必要支持其它格式，html 打印出来就是 PDF，如此以来，`output` 后面的选项是不是就极大的简化了，article、book、slider、blog、thesis、letter、card。嗯，我一直没有忘记 **pagedown**。

**Comment by yihui** [comment]
1 有些超纲，这需要编辑器支持，不是我能做的。除非还从头写个编辑器。
2 估计不会采纳。
3 4 5 都有道理。谢谢。

**Comment by chuxinyuan** [comment]
关于包名，xdown、htmldown、onedown、playdown……，当然 litedown 也很棒，我尊重你的选择。

关于函数名，tidyr 包的 `pivot_longer()` 和 `pivot_wider()` 算是改名字比较成功的。听说他是通过在推特上发起了一个投票，我觉得这办法挺好。能否将 `fuse()` 改成 `html()`，`roam()` 改成 `view()`。不过也不必太在意皮相，有仙则名。

**Comment by yihui** [comment]
就算有投票，我知道也有人痛恨这两个 `pivot_*()` 新名字。

`fuse()` 与 `html()` 完全是两码事，与 fuse() 等价的是 knit() 和 render()，它要表达的是代码和文档混编，而且并非唯一输出 html。

`view()` 太宽泛了，而且有别的包已经用这个名字了，并且也存在 `View()` 函数。

包名 litedown 对我来说唯一的小问题是八个字母太长，除此之外几乎完美。

包和函数的命名问题我已经花了大量时间去研究和琢磨，重命名的可能性应该不大了。
